[stable8] Keep versions when deleting as recipient

PVince81 opened the pull request:
Backport of https://github.com/owncloud/core/pull/16691 to stable8
- [ ] investigate whether `retainEncryptionKeys` requires a similar fix

Commit:
Keep versions when deleting as recipient

The original owner could not be resolved because the original file was
already deleted, so no versions were retained.

This fix passes the owner and path to retainVersions() since it was
already resolved before the file deletion. This reactivates the versions
logic and creates a copy of the versions in each trashbin.

nickvergessen commented on apps/files_trashbin/tests/storage.php:
```diff
@@ -187,6 +187,10 @@ public function testDeleteVersionsOfFile() {
+
+		// versions deleted
+		$results = $this->rootView->getDirectoryContent($this->user . '/files_versions/');
+		$this->assertEquals(0, count($results));
```
Please use `assertCount()`

PVince81 replied:
Damn you copy paste!!!

PVince81 replied:
I'll leave this for now. As this is a backport, I'd like to keep the difference from master low.
But I'll keep it in mind for my next work on master :wink:

nickvergessen replied:
Right, didn't see that in the beginning

nickvergessen commented on apps/files_trashbin/tests/storage.php:
```diff
@@ -218,6 +222,118 @@ public function testDeleteVersionsOfFolder() {
+
+		// check if versions are in trashbin for both users
+		$results = $this->rootView->getDirectoryContent($this->user . '/files_trashbin/versions');
+		$this->assertEquals(1, count($results), 'Versions in owner\'s trashbin');
```
OCA\Files_trashbin\Tests\Storage\Storage::testDeleteVersionsOfFileAsRecipient
Versions in owner's trashbin
Failed asserting that 0 matches expected 1.

nickvergessen commented on apps/files_trashbin/tests/storage.php:
```diff
@@ -218,6 +222,118 @@ public function testDeleteVersionsOfFolder() {
+
+		// check if versions are in trashbin for owner
+		$results = $this->rootView->getDirectoryContent($this->user . '/files_trashbin/versions');
+		$this->assertEquals(1, count($results));
```
OCA\Files_trashbin\Tests\Storage\Storage::testDeleteVersionsOfFolderAsRecipient
Failed asserting that 0 matches expected 1.

PVince81 commented:
Looks like the share mounts aren't mounted at all on stable8, the MountProviderCollection is empty. The mechanism is probably different on master now.

Needs further research to make it work.

Commit:
Register sharing hooks in trashbin tests

Register sharing hooks to make sure that the share mount points are
mounted properly, as required by the share-related trashbin tests.

PVince81 commented:
Added sharing hooks in tests to make sure the share mount points are mounted: 70bcb1a1c1b0480eff87cf171287b5095eea9ef2
This was not needed on master because share mount points are mounted differently, using mount providers.

PVince81 commented:
@nickvergessen @schiesbn @MorrisJobke please review this backport

nickvergessen commented:
:+1: 

>  - [ ] investigate whether `retainEncryptionKeys` requires a similar fix

Did you have a change for this?

PVince81 commented:
Didn't have time... still on my todo list for some time later... would be nice if @schiesbn could check it if he has time

DeepDiver1975 commented:
> Didn't have time... still on my todo list for some time later... would be nice if @schiesbn could check it if he has time

postpone to 8.0.6 and mark as developing? @PVince81 :question: THX

PVince81 commented:
I still think we can merge this. The retainEncryptionKey is completely separate. It's just something I spotted while doing this backport. (a part that is not present on master)

DeepDiver1975 commented:
thx @PVince81 :+1:
